### PR TITLE
Another drops from old testsuite

### PR DESCRIPTION
**Pull request by mchf**

### CODE REVIEW

File: src/include/network/lan/udev.rb [hunk 1]
```diff
@@ -71,10 +72,9 @@ def update_udevd
 
     # Removes (key,operator,value) tripplet from given udev rule.
     def RemoveKeyFromUdevRule(rule, key)
-      rule = deep_copy(rule)
-      pattern = Builtins.sformat("%1={1,2}[^[:space:]]*", key)
+      pattern = /#{key}={1,2}[^[:space:]]*/
```
Comment (jreidinger) [inline src/include/network/lan/udev.rb]: usually safer is to use `Regexp.escape(key)`

Comment (jreidinger) [inline src/include/network/lan/udev.rb]: also this part `[^[:space:]]` doesn't look like ruby regexp. What it actually should catch? something like `\S` ?

File: src/include/network/lan/udev.rb [hunk 2]
```diff
@@ -82,13 +82,11 @@ def RemoveKeyFromUdevRule(rule, key)
     # Tripplet is given as a string in form KEY="VALUE" or
     # MATCHKEY=="MATCHVALUE"
     def AddToUdevRule(rule, tripplet)
-      rule = deep_copy(rule)
-      if !Builtins.regexpmatch(tripplet, ".+={1,2}\".*\"")
-        return deep_copy(rule)
-      end
+      return rule unless tripplet =~ /.+={1,2}\".*\"/
+
       rule = [] if rule.nil?
```
Comment (jreidinger) [inline src/include/network/lan/udev.rb]: better `rule ||= []`

File: src/include/network/lan/udev.rb [hunk 3]
```diff
@@ -82,13 +82,11 @@ def RemoveKeyFromUdevRule(rule, key)
...
 
-      Builtins.add(rule, tripplet)
+      rule << tripplet
```
Comment (jreidinger) [inline src/include/network/lan/udev.rb]: be aware that here you return reference to rule passed as argument, so it slightly change behavior. So just ensure it do not break anything. Problem can be e.g. something like this

```
RULE="c=d"

# old behavior
res = AddToUdevRule(RULE, "a=b")
res = AddToUdevRule(RULE, "a=b")
# res will be "c=d c=d a=b" and RULE will be "c=d"
# in new behavior res and RULE will be "c=d c=d a=b a=b"
```

**Comment by jreidinger** [comment]
in general nice improvement, only few minor notes and one possible cave eat :+1:

File: src/include/network/lan/udev.rb (hunk 3 above)
Comment (mchf) [inline src/include/network/lan/udev.rb]: thanks for pointing that.I was unsure of this change. I've already checked all occurrences in the network code and it should work fine. However as I think of it now, I'll return to old behavior as it is probably more useful.

Comment (jreidinger) [inline src/include/network/lan/udev.rb]: old behavior is `rule + [tripplet]`

Comment (mchf) [inline src/include/network/lan/udev.rb]: I know, I'll update it. Thanks

**Comment by jreidinger** [comment]
in general LGTM

**Pull request merged**